ldap/servers/plugins/uiduniq/uid.c

```diff
@@ -793,7 +793,7 @@ searchAllSubtrees(Slapi_DN **subtrees, Slapi_DN **exclude_subtrees, const char *
            * violate constraint
            */
         for (i = 0; subtrees && subtrees[i]; i++) {
-            if (slapi_sdn_issuffix(dn, subtrees[i])) {
+            if (slapi_sdn_issuffix(parentDN, subtrees[i])) {
```

Comment:
Is this change required for the fix ?
If I am not wrong I think that in the case that target DN (destination entry) is one of the subtrees, then testing the parent will result in_a_subtree=False, although it should be True.

It is a corner case but may be valuable if the moved entry is the head of a complete subtree.

Reply:
Okay, I think I should rename it all as it brings a lot of confusion. :)

It's actually sourceDN and destinationDN. So for modrdn we should use destinationDN for any plugin checks as it's where we should apply the policies.
And sourceDN should be used for ignoring in the 'search()' function.